### PR TITLE
Phantomjs path fix

### PR DESCRIPTION
**Pull request by wolfgangmeyers**

### CODE REVIEW

File: third_party/phantomjs/phantomjs.sh [hunk 1]
```diff
@@ -26,6 +26,13 @@ if [[ "${RUNFILES}" == "" ]]; then
   fi
 fi
 
+# Detect if phantomjs is running within rules_closure or has been imported
+# into another project
+PHANTOMJS_BIN_PATH="third_party/phantomjs/bin/phantomjs"
```
Comment (jart) [inline third_party/phantomjs/phantomjs.sh]: Can `PHANTOMJS` be the full path of the PhantomJS binary?

File: third_party/phantomjs/phantomjs.sh [hunk 2]
```diff
@@ -26,6 +26,13 @@ if [[ "${RUNFILES}" == "" ]]; then
...
+if ! [ -e ${RUNFILES}/${PHANTOMJS_BIN_PATH} ]; then
```
Comment (jart) [inline third_party/phantomjs/phantomjs.sh]: Prefer `[[ ... ]] ` https://google.github.io/styleguide/shell.xml#Test,_%5B_and_%5B%5B

File: third_party/phantomjs/phantomjs.sh [hunk 3]
```diff
@@ -26,6 +26,13 @@ if [[ "${RUNFILES}" == "" ]]; then
...
+	PHANTOMJS_BIN_PATH=external/io_bazel_rules_closure/${PHANTOMJS_BIN_PATH}
```
Comment (jart) [inline third_party/phantomjs/phantomjs.sh]: Two space indent.

File: third_party/phantomjs/phantomjs.sh [hunk 4]
```diff
@@ -26,6 +26,13 @@ if [[ "${RUNFILES}" == "" ]]; then
...
+fi
+
 export LD_LIBRARY_PATH="${RUNFILES}/third_party/fontconfig/k8:${LD_LIBRARY_PATH}"
```
Comment (jart) [inline third_party/phantomjs/phantomjs.sh]: Why isn't it necessary to use the external/io_bazel_rules_closure prefix on these too?

**Comment by jart** [comment]
I finally understand what's happening now.

I'm not sure if this is the correct way to solve this. I'm going to take a crack at it later today if I have time.

**Comment by wolfgangmeyers** [comment]
Ok, I will hold off on pushing any more changes. Thanks!

**Comment by jart** [comment]
Does #33 fix this problem?

**Comment by wolfgangmeyers** [comment]
@jart no. I ran the reproduction scenario against the latest and get this error:
```
test/test: 2: exec: third_party/phantomjs/phantomjs.sh: not found
```

**Comment by wolfgangmeyers** [comment]
@jart I spoke too soon before updating the rules_closure import to HEAD. Now there is a new error - looks like this fails before the test is run, so I can't confirm if it's resolved or not:
```
ERROR: missing input file '@closure_library//:third_party/closure/goog/jpeg_encoder/jpeg_encoder_basic.js'.
ERROR: /home/wolfgang/.cache/bazel/_bazel_wolfgang/f22120bc19a76a8a70bf3df23fe3faac/external/io_bazel_rules_closure/third_party/noto/BUILD:21:1: Executing genrule @io_bazel_rules_closure//third_party/noto:hinted failed: namespace-sandbox failed: error executing command /home/wolfgang/.cache/bazel/_bazel_wolfgang/f22120bc19a76a8a70bf3df23fe3faac/phantomjs-path-issue/_bin/namespace-sandbox ... (remaining 5 argument(s) skipped).
ERROR: /home/wolfgang/.cache/bazel/_bazel_wolfgang/f22120bc19a76a8a70bf3df23fe3faac/external/io_bazel_rules_closure/closure/library/BUILD:20:1: @io_bazel_rules_closure//closure/library:library: missing input file '@closure_library//:third_party/closure/goog/jpeg_encoder/jpeg_encoder_basic.js'.
Target //test:test failed to build
Use --verbose_failures to see the command lines of failed build steps.
ERROR: /home/wolfgang/.cache/bazel/_bazel_wolfgang/f22120bc19a76a8a70bf3df23fe3faac/external/io_bazel_rules_closure/closure/library/BUILD:20:1 1 input file(s) do not exist.
INFO: Elapsed time: 8.122s, Critical Path: 1.54s

Executed 0 out of 1 tests: 1 fails to build.
```

**Comment by wolfgangmeyers** [comment]
disregard the last error, that was from a stale BUILD file. I've re-run, and we're back to the same issue:
```
test/test: 2: exec: third_party/phantomjs/phantomjs.sh: not found
```

**Comment by hochhaus** [comment]
@wolfgangmeyers I took a stab at fixing this in #44. Does this fix your issue?

**Comment by wolfgangmeyers** [comment]
I'll check when I have time. Thanks!

**Comment by jart** [comment]
The issues you were experiencing earlier should be resolved now.

**Pull request closed without merging**